File: LeetCode/C++/1-99/66.cpp
#include <iostream>
#include <vector>
using namespace std;
 
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int curAddDigit = 1;
        int nextAddDigit = 0;
        vector<int> ans(digits.size());
        for(int i = digits.size() - 1;i >= 0;--i){
            ans[i] = digits[i];
        }
        for(int i = digits.size() - 1;i >= 0;--i){
            nextAddDigit = (curAddDigit + ans[i]) / 10;
            ans[i] = (curAddDigit + ans[i]) % 10;
            curAddDigit = nextAddDigit;
            if(curAddDigit == 0){
                break;
            }
        }
 
        if(nextAddDigit == 1){
            ans.insert(ans.begin(), 1);
        }
 
        return ans;
    }
};
 
int main() {
	Solution sol;
	vector<int> digits{1,2,3};
	vector<int> ans = sol.plusOne(digits);
	for(int num : ans){
		cout << num << " ";
	}
	return 0;
}

File: LeetCode/C#/1-99/66.cs
using System;
using System.Collections.Generic;
namespace LeetCode66
{
	public class Solution {
	    public int[] PlusOne(int[] digits) {
	        int curAddDigit = 1;
	        int nextAddDigit = 0;
	        List<int> ans = new List<int>(digits);
 
	        for(int i = digits.Length - 1;i >= 0;--i){
	            nextAddDigit = (curAddDigit + ans[i]) / 10;
	            ans[i] = (curAddDigit + ans[i]) % 10;
	            curAddDigit = nextAddDigit;
	            if(curAddDigit == 0){
	                break;
	            }
	        }
 
	        if(nextAddDigit == 1){
	            ans.Insert(0, 1);
	        }
 
	        return ans.ToArray();
	    }
	}
 
	public class Program
	{
		public static void Main()
		{
			Solution sol = new Solution();
			int[] digits = new int[]{1,2,3};
			int[] ans = sol.PlusOne(digits);
			foreach(int num in ans){
				Console.Write(" " + num);
			}
 
			Console.Read();
		}
	}
}